### DISCUSSION TITLE
Error Column

### DISCUSSION THREAD

**Discussion by tatuananhnb2608** [rappasoft/laravel-livewire-tables#1255] [Q&A] [answered]
I can only use Column::make, the rest BooleanColumn or Image also say Class tag "App\Http\Livewire\BooleanColumn" not found
whether I loaded use Rappasoft\LaravelLivewireTables\Views\BooleanColumn;

**Comment by lrljoe** [comment]
You need to add the relevant column types in within your Use statements.

i.e.

```php
<?php

namespace App\Http\Livewire;

use Illuminate\Database\Eloquent\Builder;
use Rappasoft\LaravelLivewireTables\DataTableComponent;
use Rappasoft\LaravelLivewireTables\Views\Column;
use Rappasoft\LaravelLivewireTables\Views\Columns\BooleanColumn;
use Rappasoft\LaravelLivewireTables\Views\Columns\ImageColumn;
```

**Comment by tatuananhnb2608** [comment]
It worked but when I want to show the image from the database it can't because $row->id is not the data of the table. When I use the code " Column::make('profile_photo_path')
                ->format(function(User $model) {
                    return $this->image($model->profile_photo_path, $model->name, ['class' => 'img-fluid']);
                })" it generates the error App\Http\Livewire\UserTable::App\Http\Livewire\{closure}(): Argument #1 ($model) must be of type App\Models\User, string given

**Reply by lrljoe** [reply, marked as answer]
Yep, model is a string.

You can use setAdditionalSelects in your configure() method to add extra fields in that you want to use in labels/formats.

e.g.
```php
        $this->setPrimaryKey('id')
            ->setAdditionalSelects(['users.id as id', 'users.profile_photo_path as profile_photo_path']);
```
Then you can reference those fields in your labels/formats under the $row->fieldName approach

**Comment by tatuananhnb2608** [comment]
How wonderful. it worked thanks for all, Do you have a link to buy a coffee?

**Reply by lrljoe** [reply]
There should be a "buy me a beer" link on the readme for the package.

It's Anthony's package, and he deserves it. I'm just lending a hand where I can!

If you get stuck on anything else, or want some pointers, then feel free to open a discussion with your component code and I'll see what I can do.

**Comment by tatuananhnb2608** [comment]
Well I did it for Antony and thank you very much. I'm trying to find the use "if else" to give a message at will.If you know about it can you please tell me about it.

**Reply by lrljoe** [reply]
If you join the new Discord for this package then it may be easier to go back/forth on that!
[https://discord.gg/2d35xWxt](https://discord.gg/2d35xWxt)
